packages/side-nav/src/SideNavItem.tsx

```diff
@@ -191,23 +185,28 @@ const SideNavItem: ExtendableBox<
     ariaCurrentValue = AriaCurrentValue.Page,
     className,
     children,
+    onClick: onClickProp,
+    glyph,
+    ...rest
   } = props;
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
 
-  const rest = omit(props, [
-    'active',
-    'disabled',
-    'ariaCurrentValue',
-    'className',
-    'children',
-  ]);
+  const onClick = disabled
+    ? (e: React.MouseEvent) => {
+        e.nativeEvent.stopImmediatePropagation();
+        e.preventDefault();
```

Comment:
Why do we need to preventDefault here?

Reply:
That stops it from redirecting you if it's a link that's clicked.